src/merkle/tiered_smt/mod.rs

```diff
@@ -389,6 +417,13 @@ impl Default for TieredSmt {
     }
 }
 
+/// Data structures given to the VM's advice provider in order to perform a series of key inserts in
+/// the Miden VM
+pub struct TieredSmtInsertWitness {
+    pub partial_mt: PartialMerkleTree,
+    pub advice_map: BTreeMap<RpoDigest, Vec<(RpoDigest, Word)>>,
+}
```

Comment:
Following [this comment](https://github.com/0xPolygonMiden/miden-node/pull/59#discussion_r1381134316), I wonder if instead of returning a `PartialMerkleTree` we return just `Vec<MerklePath>`. Then, this struct could look something like this:
```Rust
pub struct TieredSmtInsertWitness {
    pub paths: Vec<MerklePath>,
    pub indexes: Vec<NodeIndex>,
    pub leaves: BTreeMap<RpoDigest, Vec<(RpoDigest, Word)>>,
}
```